apis/pipelines/v1alpha5/runconfiguration_types.go

```diff
@@ -7,29 +7,39 @@ import (
 )
 
 type Trigger struct {
-	Type TriggerType `json:"type"`
-	// +kubebuilder:validation:Required
-	CronExpression string `json:"cronExpression"`
+	Schedule *ScheduleTrigger `json:"schedule,omitempty"`
+	OnChange *OnChangeTrigger `json:"onChange,omitempty"`
 }
 
-type TriggerType string
+type ScheduleTrigger struct {
+	CronExpression string `json:"cronExpression"`
```

Comment:
Not too important though.
```suggestion
	// +kubebuilder:validation:Required
	CronExpression string `json:"cronExpression"`
```